users/dierkes/preprocessing/segment_cutting.py

```diff
@@ -0,0 +1,86 @@
+import soundfile as sf
+import numpy as np
+import multiprocessing
+import logging
+
+from sisyphus import Job, Task, tk
+
+from i6_core.lib import corpus
+
+class CutAndStitchSpeechSegmentsFromCorpusJob(Job):
+    """
+    This Job uses a bliss corpus to cut all segments containing speech from the audio files
+    and stitches them together to files of a target length. Segments that are
+    longer than the target_length are stored as individual files.
+
+
+    """
+    def __init__(self, bliss_corpus_file, target_length=20, n_workers=16, file_extension='wav'):
```

Comment:
`n_workers` should not be hashed, as this has no effect on the output of the job. You can check other jobs (e.g. ReturnnTraining or so) on how to write custom hash functions.

Reply:
@JackTemaki also suggested offline to reduce the default `n_workers` to 4 to reduce the file server load.

Reply:
```suggestion
    def __init__(self, bliss_corpus_file, target_length=20, n_workers=4, file_extension='wav'):
```

Reply:
I am not sure about this change. Even if we are only using 4 cpus we can benefit of more workers running on multiple cores of the cpus, can't we? Is this also problematic because of the server load?

Reply:
The `cpu` entry for rqmt is about the cores (or more exactly, allowed number of simultaneously running threads), not about the number of actual CPUs. It is just that you do not want to make the job so fast that you have IO as bottleneck.

Reply:
So of you specify more workers than you have as allowed threads, you program will actually be slower and not faster, because there is an increased overhead in thread swapping.

Reply:
Then we should also change `cpu` in `rqmt` to `n_workers`, right?